components/nodemanager-service/pgdb/filtering.go

```diff
@@ -84,7 +101,15 @@ func buildWhereFilter(mergeableFilters []*common.Filter, tableAbbrev string, fil
 		conditions = append(conditions, newCondition)
 	}
 
-	whereFilter = fmt.Sprintf("WHERE (%s)", strings.Join(conditions, " AND "))
+	if len(oRConditionsForTags) == 0 {
+		whereFilter = fmt.Sprintf("WHERE (%s)", strings.Join(conditions, " AND "))
+	} else {
+		if len(conditions) > 0 {
```

Comment:
is this too messy? is there a cleaner way?

Reply:
The logic is getting a bit convoluted with special cases. I think you could simplify somewhat by separating all the tag handling from the other filters. If you moved all the tag filter handling first, that stuff could be `OR`ed together and added to the `conditions` as just another element, since those all get `AND`ed together at the end.

Hopefully that made sense, I was trying to avoid just writing a bunch of pseudocode.